azurerm/internal/services/synapse/synapse_workspace_resource.go

```diff
@@ -642,3 +660,14 @@ func flattenIdentityControlSQLSettings(settings synapse.ManagedIdentitySQLContro
 
 	return false
 }
+
+func flattenEncryptionDetails(encryption *synapse.EncryptionDetails) string {
+	if cmk := encryption.Cmk; cmk != nil {
+		if key := cmk.Key; key != nil {
+			if keyVaultUrl := key.KeyVaultURL; keyVaultUrl != nil {
+				return *keyVaultUrl
+			}
+		}
+	}
+	return ""
```

Comment:
If the return value is a pointer this can change to `nil` instead of an empty string.